File: element-common/src/main/java/io/github/solusmods/eternalcore/element/impl/network/ElementsNetwork.java
package io.github.solusmods.eternalcore.element.impl.network;

import io.github.solusmods.eternalcore.element.impl.network.c2s.SyncElementsStoragePayload;
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ElementsNetwork {

    public static void init() {
        NetworkUtils.registerC2SPayload(SyncElementsStoragePayload.TYPE, SyncElementsStoragePayload.STREAM_CODEC, SyncElementsStoragePayload::handle);
    }
}

File: element-common/src/main/kotlin/io/github/solusmods/eternalcore/element/impl/network/ElementsNetwork.kt
package io.github.solusmods.eternalcore.element.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.element.impl.network.c2s.SyncElementsStoragePayload
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils
import lombok.AccessLevel
import lombok.NoArgsConstructor

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object ElementsNetwork {
    @JvmStatic
    fun init() {
        NetworkUtils.registerC2SPayload<SyncElementsStoragePayload?>(
            SyncElementsStoragePayload.Companion.TYPE,
            SyncElementsStoragePayload.Companion.STREAM_CODEC,
            NetworkManager.NetworkReceiver { obj: SyncElementsStoragePayload?, context: NetworkManager.PacketContext? ->
                obj!!.handle(context!!)
            })
    }
}
